File: src/client/java/set/starlev/starredheltix/render/RenderEvents.java
package set.starlev.starredheltix.render;

import java.util.ArrayList;
import java.util.List;

public class RenderEvents {
    private static final List<WorldRenderCallback> callbacks = new ArrayList<>();
    
    public static void register(WorldRenderCallback callback) {
        callbacks.add(callback);
    }
    
    public static void fireWorldRender(RenderContext context) {
        for (WorldRenderCallback callback : callbacks) {
            callback.onWorldRender(context);
        }
    }
    
    @FunctionalInterface
    public interface WorldRenderCallback {
        void onWorldRender(RenderContext context);
    }
}

File: src/main/kotlin/set/starlev/render/RenderEvents.kt
package set.starlev.render

class RenderEvents private constructor() {
    companion object {
        private val callbacks = mutableListOf<(RenderContext) -> Unit>()

        @JvmStatic
        fun register(callback: (RenderContext) -> Unit) {
            callbacks.add(callback)
        }

        @JvmStatic
        fun fireWorldRender(context: RenderContext) {
            for (callback in callbacks) {
                callback(context)
            }
        }
    }
}